js/controller/ContentController.js

```diff
@@ -18,14 +17,36 @@ app.controller('ContentController', function (Publisher, FeedResource, ItemResou
     // distribute data to models based on key
     Publisher.publishAll(data);
 
+    var getOrdering = function () {
+        var ordering = SettingsResource.get('oldestFirst');
+
+        if (self.isFeed()) {
+            var feed = FeedResource.getById($routeParams.id);
+            if (feed && feed.ordering === 1) {
+                ordering = true;
+            } else if (feed && feed.ordering === 2) {
+                ordering = false;
+            }
+        }
+
+        return ordering;
+    };
+
     this.getFirstItem = function () {
```

Comment:
This function can't call `this.orderBy` anymore, but there's an easier way of finding what the first item should be. Just check `getOrdering` and then pick the first or last item in the list accordingly.